javascript/modules/client/src/client-erc20.ts

```diff
@@ -434,21 +432,140 @@ export class ClientErc20 extends ClientCore implements IClientErc20 {
+      // format in the metadata field
+      const test_cid = "QmXhJawTJ3PkoKMyF3a4D89zybAHjpcGivkb7F1NkHAjpo";
+      const metadataString = await this.ipfs.fetchString(test_cid);
+      const metadata = JSON.parse(metadataString) as ProposalMetadata;
+      const startDate = new Date(
+        parseInt(erc20VotingProposal.startDate) * 1000,
+      );
+      const endDate = new Date(parseInt(erc20VotingProposal.endDate) * 1000);
+      const creationDate = new Date(
+        parseInt(erc20VotingProposal.createdAt) * 1000,
+      );
+      let usedVotingWeight: bigint = BigInt(0);
+      erc20VotingProposal.voters.forEach(
+        (voter: { voter: { id: string }; vote: string; weight: string }) => {
+          usedVotingWeight = usedVotingWeight + BigInt(voter.weight);
+        },
+      );
+      // get status
+      const status = getProposalStatus(
+        startDate,
+        endDate,
+        erc20VotingProposal.executed,
+        erc20VotingProposal
+          .yea,
+        erc20VotingProposal.nay,
+      );
+      // get start and end date
+      return {
+        id: erc20VotingProposal.id,
+        dao: {
+          address: erc20VotingProposal.dao.id,
+          name: erc20VotingProposal.dao.name,
+        },
+        creatorAddress: erc20VotingProposal.creator,
+        metadata: {
+          title: metadata.title,
+          summary: metadata.summary,
+          description: metadata.description,
+          resources: metadata.resources,
+          media: metadata.media,
+        },
+        startDate,
+        endDate,
+        creationDate,
+        actions: erc20VotingProposal.actions.map(
+          (action: { data: string; to: string; value: string }): DaoAction => {
+            return {
+              data: hexToBytes(strip0x(action.data)),
+              to: action.to,
+              value: BigInt(action.value),
+            };
+          },
+        ),
+        status,
+        result: {
+          yes: BigInt(erc20VotingProposal.yea),
+          no: BigInt(erc20VotingProposal.nay),
+          abstain: BigInt(erc20VotingProposal.abstain),
+        },
+        settings: {
+          // TODO
+          // this should be decoded using the number of decimals that we want
+          // right now the encoders/recoders use 2 digit precission but the actual
+          // subgraph values are 18 digits precision. Uncomment below for 2 digits
+          // precision
+
+          // minSupport: decodeRatio(
+          //   BigInt(erc20VotingProposal.supportRequiredPct),
+          //   2,
+          // ),
+          // minTurnout: decodeRatio(
+          //   BigInt(erc20VotingProposal.participationRequiredPct),
+          //   2,
+          // ),
+          // TODO DELETE ME
+          minSupport: parseFloat(
+            formatEther(erc20VotingProposal.supportRequiredPct),
+          ),
+          minTurnout: parseFloat(
+            formatEther(erc20VotingProposal.participationRequiredPct),
+          ),
```

Comment:
Aren't these percent values? We have the helper for encoding/decoding float's as bigints. Not sure why Ethers formatting is needed here

Reply:
Yes, there is a big TODO explaining why this, but basically till now the `minTurnout` and `minSupport` were stored as `BigInts` with 18 numbers of precession. Right now we are storing them with a 2 digit precession, and our encoders-decoders have a max of 15 numbers of precision, so we will need to use the `formatEtther` until we start using the new contracts and storing them with 2 digit precision

Reply:
Subgraph, at this moment, has a mix of different ways of formatting this values so until we have a standard multiple nonsense parse will happen depending on the situation